tools/CheckDates.scala

```diff
@@ -0,0 +1,184 @@
+    var rootFolder = findRootFolder()
+    val myval =  (s"find ${rootFolder} -name *.scala ! -path ./tools/*  -type f") !!
+    val files_array = myval.split("\n").map(_.toString).distinct
+    files_array
+  }
+
+  def readFile(pathToFile: String): String = {
+    val source = scala.io.Source.fromFile(pathToFile)("UTF-8")
+    val lines = try source.mkString finally source.close()
+    lines
+  }
+
+  def ifFileWasInGit(pathToFile: String): Boolean = {
+    val res_dates = (s"git log $pathToFile" !!)
+    if(res_dates == "") {
+      false
+    } else {
+      true
+    }
+  }
+
+  def getGitCreationYear(pathToFile: String): String = {
+    val buf_dates = (s"git log   --format=%cd $pathToFile" !!)
+    val dates = buf_dates.split("\\n").distinct
+    val res = dates.last.split(" ")(4)
+    res
+  }
+
+  def getGitLastUpdateYear(pathToFile: String): String = {
+    val buf_dates = (s"git log -3  --format=%cd $pathToFile" !!)
+    val buf_messages = (s"git log -3  $pathToFile" !!)
+    val messages = buf_messages.split("commit").distinct.drop(1)
+    val dates = buf_dates.split("\\n").distinct
+    val gits_info = (messages, dates).zipped.map{ case(a,b) => GitInfo.tupled((a,b))}
+    val res = gits_info.filter(!_.message.contains("[Dates fixing]"))
+    res(0).date.split(" ")(4)
+  }
+
+  def getYear(pathToFile: String, lower_bound: Int, upper_bound: Int): String = {
+    val file = readFile(pathToFile)
+    val start_index = file.indexOf("(c) 20")
+    val year = file.substring(start_index + lower_bound, start_index + upper_bound)
+    year
+  }
+
+  def getActualCreationYear(pathToFile: String): String = {
+    val lb = "(c) ".length()
+    val ub = "(c) 20XX".length()
+    val actualCreationYear = getYear(pathToFile, lb, ub)
+    actualCreationYear
+  }
+
+  def getActualLastUpdateYear(pathToFile: String): String = {
+    val lb = "(c) 20XX-".length()
+    val ub = "(c) 20XX-XXXX".length()
+    val actualLastUpdateYear = getYear(pathToFile, lb, ub)
+    actualLastUpdateYear
+  }
+
+  def analyzeDates(): Array[FileDates] = {
+    val files_array = readFilesList()
+    var data = Array[FileDates]()
+    for(file_path <- files_array) {
+      if(readFile(file_path).contains("|    (c) 20") && ifFileWasInGit(file_path)) {
+        val actualLastUpdateYear = getActualLastUpdateYear(file_path)
+        val gitLastUpdateYear = getGitLastUpdateYear(file_path)
+        val actualCreationYear = getActualCreationYear(file_path)
+        val gitCreationYear = getGitCreationYear(file_path)
+        data = data :+ FileDates.tupled((file_path,
+                        actualLastUpdateYear,
+                        gitLastUpdateYear,
+                        actualCreationYear,
+                        gitCreationYear))
+      }
+    }
+    data
+  }
+
+  def fixWrongDates(): Boolean = {
+    var check = false
+    val files = analyzeDates()
+    files.foreach { file =>
+      val actual_file = readFile(file.pathToFile)
+      var new_file = ""
```

Comment:
Using `""` as a rogue value is fine but you could consider using `None`/`Some` instead.